lib/cheetah.rb

```diff
@@ -455,54 +467,70 @@ def build_recorder(options)
       end
     end
 
-    def fork_commands_recursive(commands, pipes)
+    # Reopen *stream* to write **into** the writing half of *pipe*
+    # and close the reading half of *pipe*.
+    # @param pipe [Array<IO>] a pair of IOs as returned from IO.pipe
+    # @param stream [IO]
+    def into_pipe(stream, pipe)
```

Comment:
My reasoning was to make it like `stream.into_pipe(pipe)`. I am not aware of a pattern that would make one way preferred over the other. Anyway, it is a simple private method, so I'd leave it as it is.